skops/card/_model_card.py

```diff
@@ -410,9 +410,12 @@ def _generate_card(self) -> ModelCard:
                         'clf.predict(pd.DataFrame.from_dict(config["sklearn"]["example_input"]))'
                     )
         if self.model_diagram is True:
-            model_plot: str | None = re.sub(
+            model_plot_div = re.sub(
                 r"\n\s+", "", str(estimator_html_repr(self.model))
             )
+            assert model_plot_div.count("sk-top-container") == 1
```

Comment:
production code should never have `assert` in it. Instead, you can check the value, and handle the case where it doesn't satisfy the condition.